src/web/app/pages-instructor/instructor-search-page/comment-result-table/comment-result-table.component.ts

```diff
@@ -0,0 +1,30 @@
+import { Component, Input, OnInit } from '@angular/core';
+import { CommentTableModel } from "../../../components/comment-box/comment-table/comment-table.component";
+import { SearchCommentsTable } from "../instructor-search-page.component";
+import { ResponseOutput } from "../../../../types/api-output";
+
+export interface CommentListSectionData {
```

Comment:
Is it necessary to define the interface here? The structure of API output is far enough?

Reply:
Alright, that makes sense. We can just keep the feedback session and question output API outputs since all the fields are there. But I think there should be an additional commentTableModel as the comment table needs this model as data. The transformation is then done in the search service itself. So we can just have 
```
SearchCommentsTable {
    feedbackSession: FeedbackSession,
    questions: QuestionOutput[],
    commentTableModel: CommentTableModel
}
```
Sounds like a better interface?

Reply:
> But I think there should be an additional commentTableModel as the comment table needs this model as data.

The examples in the code base is actually not good. Service cares about business logic so it should not transform how data should be presented.

Regarding how to transform instructor comment to comment table model, you can use pipe to do so.

Reply:
Alright, then leave the data transformation to the component displaying it then

Reply:
FYI

> We can just keep the feedback session and question output API outputs since all the fields are there.

This is usually true if it is just a component for data presentation. See here or results page for reference.

However, if it involves form inputs (e.g. user can enter something), usually we will need to transform data to a *Model (see question edit form for reference) (and it is also why the comment table does not accept `FeedbackResponse` as a input).